crates/astria-composer/src/executor/mod.rs

```diff
@@ -94,7 +99,19 @@ pub(crate) use builder::Builder;
 const BUNDLE_DRAINING_DURATION: Duration = Duration::from_secs(16);
 
 type StdError = dyn std::error::Error;
-
+#[derive(Debug, thiserror::Error)]
+pub(crate) enum EnsureChainIdError {
+    // <other error variants from getting genesis, etc>
+    #[error("failed to obtain sequencer chain ID")]
+    FailedToGetChainId,
```

Comment:
Don't throw away the error:
```suggestion
    GetChainId(#[source] sequencer_client::tendermint_rpc::Error),
```
^ check if that's the right type. I think it is

Note on the variant: because this already is an error it's recommended to not repeat "failed" or "error" in the variants.